tests/testthat/test-growth_rate.R

```diff
@@ -0,0 +1,56 @@
+library(dplyr)
+
+X <- c(1:10,NA,10:20,NA)
+Y <- c(2^(1:9),NA,NA,2^(10:21))
+
+methods <- c("rel_change","linear_reg","smooth_spline","trend_filter")
+
+gr <- function(method = "rel_change", h = 3, na_rm = TRUE, ...) {
+  growth_rate(x=X,y=Y,method=method,na_rm = na_rm, h = h,...)
+}
+
+test_that("Test error throwing",{
+  # Error cases
+  expect_error(growth_rate(x=1:3,y=1:4),
+               "`x` and `y` must have the same length.")
+  expect_error(growth_rate(x=1:20,y=1:20,x0=21),
+               "`x0` must be a subset of `x`.")
+  # Fails only when method = `"trend_filter"`
+  expect_error(gr(method = "trend_filter",cv=FALSE,df=1.5),
+               "If `cv = FALSE`, then `df` must be an integer.")
+})
+
+test_that("Test throwing of warning of duplicates",{
+  # The warning that is prompted is as follows:
+  # "`x` contains duplicate values. (If being run on a column in an `epi_df`,
+  # did you group by relevant key variables?)"
+  # Note that putting it in the regexp of expect_warning doesn't seem to work
+  jhu_csse_daily_subset %>%
+    mutate(cases_gr = growth_rate(x = time_value,  y = cases, dup_rm=TRUE)) %>%
+    expect_warning() %>%
+    expect_error()
```

Comment:
issue: It's hard to tell what this is doing. If we want to check for both a warning and an error, run this twice with each `expect_*`. If we're trying to do something else, I don't understand.